packages/forma-36-react-components/src/components/Textarea/Textarea.tsx

```diff
@@ -44,87 +46,81 @@ const defaultProps: Partial<TextareaProps> = {
   willBlurOnEsc: true,
 };
 
-export class Textarea extends Component<TextareaProps, TextareaState> {
-  static defaultProps = defaultProps;
+export const Textarea = (props: TextareaProps) => {
+  const {
+    className,
+    testId,
+    placeholder,
+    maxLength,
+    onChange,
+    disabled,
+    required,
+    onBlur,
+    onKeyDown,
+    error,
+    width,
+    value,
+    name,
+    rows,
+    id,
+    willBlurOnEsc,
+    textareaRef,
+    ...otherProps
+  } = props;
+  const [valueState, setValueState] = useState<string | undefined>(value);
 
-  state = {
-    value: this.props.value,
-  };
-
-  UNSAFE_componentWillReceiveProps(nextProps: TextareaProps) {
-    if (this.props.value !== nextProps.value) {
-      this.setState({
-        value: nextProps.value,
-      });
-    }
-  }
+  useEffect(() => {
+    setValueState(value);
+  }, [value]);
 
-  handleKeyDown = (e: KeyboardEvent<HTMLTextAreaElement>) => {
+  const handleKeyDown = (e: KeyboardEvent<HTMLTextAreaElement>) => {
```

Comment:
I think you should wrap this in a `useCallback` to avoid rebinding it on every keystroke 🙂